src/TenancyServiceProvider.php

```diff
@@ -74,6 +71,10 @@ public function register(): void
         $this->app->bind('globalCache', function ($app) {
             return new CacheManager($app);
         });
+
+        if ($this->app['config']['tenancy.database.drop_tenant_databases_on_migrate_fresh']) {
+            $this->app->extend('command.migrate.fresh', fn() => new Commands\MigrateFreshOverride);
```

Comment:
I see nothing with `command.migrate.fresh` binding in the Laravel source code.

Try using `\Illuminate\Database\FreshCommand::class` instead of that for the binding.

Also, FWIW for testing get rid of the `if ()`, or at least add a `dd()` inside it to verify that it works. Sometimes config can't be read this early in the request lifecycle, sometimes you need to do it at a later point.

Reply:
I see. Thanks, I'll try the binding with `FreshCommand`

Reply:
Great, you were right, using `FreshCommand::class` instead of `command.migrate.fresh` works to override the command

Reply:
@stancl, testing that the config value enables/disables the `migrate:fresh` override is tricky because TenancyServiceProvider has to boot again for the config change to have an effect. Booting the service provider mid-test doesn't override the command. Do you have any ideas on how to test this?

Reply:
> Sometimes config can't be read this early in the request lifecycle, sometimes you need to do it at a later point.

Similarly to what I wrote here, can't we move the `if ()` check to the `extend()` closure?

If the config key is off, we'll just return the parameter passed to the closure (the original object), otherwise we'll return our own instance.